File: Fenicia.Module.Basic/Domains/Order/Responses/GetAllOrderResponse.cs
using Fenicia.Common.Enums.Basic;

namespace Fenicia.Module.Basic.Domains.Order.Responses;

/// <summary>
///     Response containing order summary for list views.
/// </summary>
public record GetAllOrderResponse(
    Guid Id,
    string OrderNumber,
    Guid UserId,
    Guid CustomerId,
    string CustomerName,
    decimal TotalAmount,
    decimal DiscountAmount,
    int TotalQuantity,
    DateTime SaleDate,
    string Status,
    PaymentMethod PaymentMethod,
    int TotalItems,
    Guid? EmployeeId = null,
    string? EmployeeName = null);

File: fenicia-web/src/types/basic/order/get-all-order-response.ts
import { OrderStatus } from "./order-status";
import { PaymentMethod } from "./payment-method";

export type GetAllOrderResponse = {
    id: string;
    orderNumber: string;
    userId: string;
    customerId: string;
    customerName: string;
    totalAmount: number;
    discountAmount: number;
    totalQuantity: number;
    saleDate: string;
    status: OrderStatus;
    paymentMethod: PaymentMethod;
    totalItems: number;
    employeeId?: string;
    employeeName?: string;
};
